E2E protocol

taoeffect opened the pull request:
- Closes #92 
- Closes #112 
- Closes #1765
- Closes #839
- Closes #852

taoeffect commented on shared/domains/chelonia/internals.js:
```diff
@@ -217,22 +472,66 @@ export default (sbp('sbp/selectors/register', {
     }
-    if (processOp && !config.skipProcessing) {
-      opFns[opT](opV)
+    opFns[opT](opV)
```
@corrideat Why was this moved out of the `if`?

taoeffect commented on shared/domains/chelonia/internals.js:
```diff
@@ -283,7 +714,7 @@ export default (sbp('sbp/selectors/register', {
       preHandleEvent && await preHandleEvent(message)
       // verify we're expecting to hear from this contract
-      if (!state.pending.includes(contractID) && !state.contracts[contractID]) {
+      if (isNaN(1) && !state.pending.includes(contractID) && !state.contracts[contractID]) {
```
Need to not forget about this

taoeffect commented on shared/domains/chelonia/internals.js:
```diff
@@ -447,11 +890,11 @@ function loadScript (file, source, hash) {
     // problem with this is that scripts will step on each other's feet
-    script.innerHTML = source
+    script.text = source
     // NOTE: this will work if the file route adds .header('Content-Type', 'application/javascript')
     // script.src = `${this.config.connectionURL}/file/${hash}`
     // this results in: "SyntaxError: import declarations may only appear at top level of a module"
-    // script.innerHTML = `(function () {
+    // script.text = `(function () {
```
Undo?

taoeffect commented on frontend/controller/actions/group.js:
```diff
@@ -56,6 +48,33 @@ async function saveLoginState (action: string, contractID: string) {
+  }
+})
+*/
+
```
Can be removed?

taoeffect commented on shared/domains/chelonia/utils.js:
```diff
@@ -0,0 +1,151 @@
+    if (key.foreignKey) {
+      const fkUrl = new URL(key.foreignKey)
+      const foreignContract = fkUrl.pathname
+      const foreignKeyName = fkUrl.searchParams.get('keyName')
```
As a user-provided parameter, can this be abused in any way?

corrideat replied:
I'm not sure that this is 'user-provided' any more than the rest of the parameters, but I guess it would result in perhaps syncing the wrong key at worst.

taoeffect commented on shared/domains/chelonia/GIMessage.js:
```diff
@@ -102,24 +159,33 @@ export class GIMessage {
   message (): Object { return this._message }
 
-  op (): GIOp { return this.message().op }
+  op (): GIOp { return this._decrypted ? [this.head().op, this.message(), this.decryptedValue()] : [this.head().op, this.message()] }
```
Why is this `this.decryptedValue()` being included in the op?

corrideat committed:
Closure for storeSecretKeys to prevent logging secrets

corrideat committed:
Remove promiseAllSettled. Bugfixes.

SebinSong committed:
add /push/send route , setup 'push' event listener in service-worker

SebinSong committed:
add theme_color, background_color to the PWA manifest

SebinSong committed:
Send test notification upon sign-up / log-in , update POST /push/send so it can send push to a particular endpoint

SebinSong committed:
remove unused route POST /push/send/{subscriptionId}

SebinSong committed:
add payload validations to /push/* routes

SebinSong committed:
remove console.log

corrideat committed:
Side-effects without await

corrideat committed:
Update okturtles.eventqueue

corrideat committed:
Fix invocation

Silver-IT committed:
Make sure we sync any identity contracts we haven't synced upon login (#1763)

* feat: sync missing identity contracts after joining group

* chore: remove IIFE

* chore: resolved feedback

corrideat committed:
Remove await on /out in internals.js

corrideat committed:
Remove queueInvocation for /out events

corrideat committed:
Merge branch 'e2e-protocol' into e2e-protocol-ricardo

taoeffect committed:
Merge pull request #1758 from okTurtles/e2e-protocol-ricardo

E2e protocol ricardo

Silver-IT committed:
fix: disable change group settings unless group creator tries

corrideat committed:
Make cannot leave if not a member error into a warning

corrideat committed:
Show login error prompt only for errors related to the identity contract itself

corrideat committed:
Fix issue of removing two members (error on second removal)

corrideat committed:
Update identity.js login logic, remove spurious sideEffect check

corrideat committed:
pin contracts

taoeffect committed:
Merge pull request #1776 from okTurtles/e2e-protocol-ricardo

E2e protocol ricardo

Silver-IT committed:
Implement a way to set default groupId (#1773)

* feat: set default group ID which is successfully joined

* fix: recovered package.json

corrideat committed:
Fix inability to join after sharer signs out

Silver-IT committed:
Updated login error message (#1731)

* feat: make login error message more informative

* chore: added a line to log error

Silver-IT committed:
Updated login error message (#1731)

* feat: make login error message more informative

* chore: added a line to log error

Silver-IT committed:
fix: error of Member Removal (#1764)

corrideat committed:
Fix infinite re-subscription due to foreign keys

taoeffect committed:
Revert "Updated login error message (#1731)"

This reverts commit cf4b7f48edf847b6ad3c9c4cf4dd832c668379b2.

corrideat committed:
Fix undefined in vmInvites

taoeffect committed:
Merge pull request #1778 from okTurtles/e2e-protocol-ricardo

E2e protocol ricardo

corrideat committed:
preSendCheck for outgoing messages; move hooks to publishEvent

corrideat committed:
Remove preSendCheckContract

corrideat committed:
Remove await from call to leaveAllChatRoomsUponLeaving

taoeffect committed:
Merge pull request #1779 from okTurtles/e2e-protocol-ricardo

E2e protocol ricardo

SebinSong committed:
merge the latest master in

SebinSong committed:
Merge remote-tracking branch 'origin/e2e-protocol' into sebin/task/926-create-pwa

SebinSong committed:
re-run after merging the latest e2e-protocol

corrideat committed:
Login sync fixes

corrideat committed:
Fix various issues related to login and re-joining

taoeffect committed:
Merge pull request #1775 from okTurtles/1707-should-block-updating-group-setting-unless-group-creator-tries

Disable to change group settings unless group creator tries

corrideat committed:
Make saveOurLoginState async

corrideat committed:
Various improvments and bugfixes

corrideat committed:
Bugfixes. Revert changes to side-effects

corrideat committed:
bump

corrideat committed:
Address re-joining of chatrooms

corrideat committed:
Address PR review feedback

SebinSong committed:
update for Ricardo's feedbabck

SebinSong committed:
Merge remote-tracking branch 'origin/e2e-protocol' into sebin/task/926-create-pwa

corrideat committed:
Deferred remove

corrideat committed:
Fixes to "chelonia/private/deleteRevokedKeys"

corrideat committed:
Re-add sync to publishEvent

corrideat committed:
Bugfix: reducer at deleteRevokedKeys

corrideat committed:
Test state.contracts[cID] instead of state[cID]

corrideat committed:
Bugfixes:

  * Pass params from sync to syncContract
  * Use fresh state in signData and encryptData
  * Check membership status in Join.vue

corrideat committed:
Move joining #General and adding group key into side-effect

corrideat committed:
Bugfixes

corrideat committed:
Clear deferredRemove flag

corrideat committed:
Check if we have the decryption key first

corrideat committed:
Updated error message

corrideat committed:
Remove debug logs and fix multiple join calls

corrideat committed:
Remove unnecessary sync of identity contract

Silver-IT committed:
Update all test scenarios for updated workflow (#1782)

* Fix issues with chatroom, move some key logic into selectors

* Rename selector to isWaitingForKeyShare

* fix: update description for group channels

* WIP - Fix manifest not found error upon contract file change (#1703)

* Fix manifest not found error on contract file change

* Mark the gruntfile as WIP

* Remove WIP tag and code duplication

* Fix Chat UI bugs (#1712)

* fix: error of scrolling outside of the ChatMain section

* feat: jump to the latest when resize window

* feat: jump to the latest message without any delay when resize window

* fix: auto scroll 25 times a second

* File attachment to chat - 1st chunk (#1694)

* add file attachment button to the chat input-area el

* dnd mixin skeleton

* remove trailing space

* create the image preview vue component.

* task 1-2. create a preview file component for a non-image attachment

* build 'File too large' modal and pop it out when size is too large

* add a compressed svg image for attachment / make sure it can be colored via css

* add links for the drag&drop references

* display overlay image when chat-area is dragged over

* hook up DnD behavior with file-attachements method

* resolving some of Greg's CR

* another chunk of work for Greg's CR

* work on Greg's CR / multiple file attachments

* fix the test failure

* make it possible to append attachments when drag&drop is performed

* resolve the linter error

* work on Greg's CR / fix the bug where dot is displayed

* Add signedData, change message format

* Re-add validation

* Fix flow errors

* Fix OP_ATOMIC handling

* Harmonise types for signedData and encryptedData

* Proposal to update distribution date (#1643)

* feat: created a DistributionDate modal

* chore: improve translation

* feat: updated translation and added isInFirstDistributionPeriod

* feat: completed the change distribution date workfloww

* feat: restrict group members to change group settings

* feat: improve updateSettings group action

* feat: exception handler when open proposal modal directly

* fix: error to close General Proposal modal

* feat: improve DistributionDate proposal workflow

* feat: proposal Item for distribution date

* feat: simplified interactive message templates

* feat: proposal to change distribution date

* fix: cypress error with distribution updates

* feat: resolved feedback from @sebin

* feat: test case for changing distribution date

* fix: travis error regarding distribution date

* fix: disable submit button instead of close proposal modal

* fix: groupDistributionStarted using Date

* feat: simplified

* fix: init scrollDistance when switch channel

* fix: simplified payment periods config

* fix: updated translation

* chore: remove DRY issue and update translation

* feat: localized texts for proposals

* Implement inner signatures for OP_KEY_REQUEST

* Inner signatures for actions

* Add wrapper to encryptedData result

* Fix flow errors

* Fix rotate keys error when content is undefined

* Fix JSON serialization issues for incoming encrypted data

* Reintroduce toJSON, fix issue with ()

* Use additional data for encryption

* Start adding user keys to group & chatrooms

* Add direction attribute to GIMessage

* Add comment

* Simplified some of the code relating to GIMessage and signedData

* Fixed reference to headJSON

* Rudimentary support for inner signatures (missing permissions)

* Various changes:

* Light refactoring of `encryptedAction` to make it clearer
  - Actions to own own identity do not use inner signatures
* Permission checking and enforcement for inner signatures
  - Now, permissions are checked in a uniform way in a central
    location
* Add new `ourContactProfilesById` and `ourContactsById` getters

* Changes for optinally encrypting OP_KEY_*

* Fix key re-selection issue

* Should not display archived proposals from other group (#1723)

* feat: destructure archPropData

* chore: reverted style changes

* chore: changed variable name

* chore: added comments

* Create ci.yml

* Rethrow uncaught Vue errors in dev mode or CI (#1708)

* Rethrow uncaught Vue errors in dev mode or CI

* Apply review

* WIP: Use encrypted OP_KEY_* and new OP_KEY_REQUEST

* Bugfixes

* Merge PrivateDM and GroupDM (#1709)

* feat: direct messages in identity contract

* feat: add/remove getters

* feat: simplified getters

* wip: merging privateDMs and groupDMs

* fix: coding issues

* feat: UI updates for the DMs with more than 3

* fix: add member to dm

* feat: changed base branch from e2e-protocol-ricardo to e2e-protocol

* fix: error in get dm by usernames

* chore: added comments

* fix: error in function parameters declaration

* fix: inconsistent naming of members when create a new DM

* fix: redundant message rendering issue

* feat: grun tpin

* chore: removed wrong version of contracts

* feat: grunt pin:0.1.8

* feat: added comments

* chore: removed changes in chelonia

* chore: added comments

* chore: fix typo

* feat: error handling

* chore: removed unnecessary comment

* feat: improved chatroom types and added comments

* feat: simplified getter

* chore: simplified

* chore: added comments, error handling

* feat: merged DMs from inside and outside of the group

* feat: remove workflow regarding the dm visibility

* Bugfixes for new OP_KEY_SHARE

* feat: remove notifications when leave group (#1722)

* Bugfixes

* Improvements & bugfixes

chelonia.js: Use regular signedOutgoingData when possible
group.js: Fix name of selector

* Bugfixes

* Bugfixes for UI error handling. Move encryptedOutgoingData into /keyDel and /keyUpdate

* Implement some PR feedback

* Make KRS+KR atomic; disable deleting state when logging out

* Fix flow issues

* Mark contract as dirty until keys are received

* Bugfix: call group join upon login

* Add comments

* Encrypted storage

* chore: resolved conflicts

* feat: giCreateGroup waits until the user joins the group

* chore: fixed some errors

* fix: error egarding Awesome button disability

* Mark contract as dirty until keys are received

* Bugfix: call group join upon login

* Add comments

* feat: group-chat-direct-message.spec.js test passed

* Remove event listener when logging out

* Use settings for key storage instead of SessionStorage

* Refactoring: state encryption logic

* Simplify check

* Update references to session storage with app settings

* Support variable distribution period length (#1691)

* Fix Vue error in groupProposalSettings

* Remove unused paymentTotalFromUserToUser getter

* Add groupSortedPeriodKeys getter in group.js

* Update getters to support variable period length

* Add isIsoString in time.js and use it

* Support variable period length in MonthOverview.vue

* Apply review

* Add periodStampsForDate in time.js

* Add getHistoricalPaymentPeriods in PaymentMixin.js

* Rename getSortedPeriodKeys to getAllSortedPeriodKeys

* Add a few comments

* Export new helpers from time.js

* Improve validation in periodStampsForDate

* Remove occurences of 'undefined'

* Fix missing break statement

* Fix wrong getter name in SupportHistory.vue

* fixup! Improve validation in periodStampsForDate

* Restore logging in Cypress

* Fix reactivity issue in MonthOverview.vue

* Fix data() in PaymentRowReceived/Sent.vue

* Use groupDistributionStarted i/o inWaitingPeriod

* Revert change in updateDistributionDate

* Use humanStart/DueDate in MonthOverview.vue

* Simplify PaymentDetail.vue/initializeDetails

* Use mounted i/o watch in PaymentRowReceived/Sent.vue

* Fix pedantic Flow error

* Pin contracts to 0.1.8

* Move mounted() near the top

* Revert humanDate to plain import in MonthOverview.vue

* Use payment.period to fix Invalid Date errors

* Add .start field in initPaymentPeriod

* Add .end field in in-memory payment periods

* Rename getPeriodPayment to getPaymentPeriod

* Fix issue 1739

* Pin contracts

* Add test for payment in 2nd period

* Fix conditions for omitting empty key ops

* EncryptedValue: ensure that it's a string

* Refactor: logout handler

* Convert password to passwordFn to prevent logging

* Fix issue (key rotation / private channel not working after rejoin)

* Update Style-Guide.md

Added `User-facing Strings Guide`

* Update Style-Guide.md

typo

* Update Style-Guide.md

Improved translation section

* Update Style-Guide.md

* Update Style-Guide.md

* Update Style-Guide.md

* Fix: leaving private chat without being member

* Update Style-Guide.md

fix typos

* Documentation

* feat: updated test cases according to the updated invitation workflow

* Update Style-Guide.md

Consistent use of colons

* #1742 Use 'submit-button' component for asynchronous actions (#1750)

* add a todo comment for future <submit-button /> replacement

* add submit-button component to various places

* work on Greg's CR

* #1704 - Add 'Export to CSV' feature to payments table (#1724)

* add export CSV button to the table

* create ExportPaymentsModal.vue and register it

* make sure all the UI side of work are ready

* complete CSV Extraction logic

* complete the download CSV file logic

* fix the linter error

* work on Greg's feedback on the button position

* make sure 'Export CSV' button is not overlapped regardless of the screen width

* fix the linter error

* put all-period option into the dropdown

* remove the unecessary extra padding

* DRY PaymentsMixin.js

* update for Greg's CR on the PR

* fix the broken translation o the modal title

* restore package.json wtf...

* Bugfixes

* Add UI prompt to login and re-order buttons

* Rename button names

* Bugfix: null inner signing key when leaving chatroom

* Add comments explaining the innerSigning attributes

* Bugfix: race condition when joining due to skipActionProcessing

* Change bad previous head log level to warn

* Remove skipActionProcessing from respondToKeyRequests

* wip: second half

* feat: added profile-card in payments table

* [e2e-protocol] Fix the welcome screen bug in /pending-approval page (#1748)

* render welcome screen as position:fixed in /pending-approval page

* make the selector higher priority

* Fixed error to save scroll position (#1730)

* feat: direct messages in identity contract

* feat: add/remove getters

* feat: simplified getters

* wip: merging privateDMs and groupDMs

* fix: coding issues

* feat: UI updates for the DMs with more than 3

* fix: add member to dm

* feat: changed base branch from e2e-protocol-ricardo to e2e-protocol

* fix: error in get dm by usernames

* chore: added comments

* fix: error in function parameters declaration

* fix: inconsistent naming of members when create a new DM

* fix: redundant message rendering issue

* feat: grun tpin

* chore: removed wrong version of contracts

* feat: grunt pin:0.1.8

* feat: added comments

* chore: removed changes in chelonia

* chore: added comments

* chore: fix typo

* feat: error handling

* chore: removed unnecessary comment

* feat: improved chatroom types and added comments

* feat: simplified getter

* chore: simplified

* chore: added comments, error handling

* fix: error to save scroll position

* chore: removed allowDMInvite

* Bugfix: key rotation on watched contracts

* Fix message format for OP_KEY_SHARE

* Rotate PEK when a group member leaves

* Generic definition for the various DB stores

* Revert changes to frontend/model/database.js and use the identity contract ID as key

* Remove withEnv

* fix: error regarding passwordFn

* bug-fix for closing modal reacting too slowly

* Handle encrypted and unencrypted invites (accounting)

* fear: remove archived data when leave the group (#1747)

* Add return

* Improvements to respondToKeyRequests

* fix: errors in cypress test

* Make sync no-op for existing contracts

* fix: cypress issues

* fix: error after login

* Listen for login event before calling postSubmit

* Await in the login form, documentation

* Updates to postSyncOps

* Split up syncContractAndWatchKeys

* Add chelonia/private/enqueuePostSyncOps to blacklist

* Closure for storeSecretKeys to prevent logging secrets

* Remove promiseAllSettled. Bugfixes.

* Update okturtles.eventqueue

* Side-effects without await

* Fix invocation

* Make sure we sync any identity contracts we haven't synced upon login (#1763)

* feat: sync missing identity contracts after joining group

* chore: remove IIFE

* chore: resolved feedback

* Remove await on /out in internals.js

* Remove queueInvocation for /out events

* Make cannot leave if not a member error into a warning

* Show login error prompt only for errors related to the identity contract itself

* Fix issue of removing two members (error on second removal)

* Update identity.js login logic, remove spurious sideEffect check

* pin contracts

* Implement a way to set default groupId (#1773)

* feat: set default group ID which is successfully joined

* fix: recovered package.json

* Fix inability to join after sharer signs out

* Updated login error message (#1731)

* feat: make login error message more informative

* chore: added a line to log error

* fix: error of Member Removal (#1764)

* Fix infinite re-subscription due to foreign keys

* Revert "Updated login error message (#1731)"

This reverts commit cf4b7f48edf847b6ad3c9c4cf4dd832c668379b2.

* Updated login error message (#1731)

* feat: make login error message more informative

* chore: added a line to log error

* Fix undefined in vmInvites

* preSendCheck for outgoing messages; move hooks to publishEvent

* Remove preSendCheckContract

* Remove await from call to leaveAllChatRoomsUponLeaving

---------

Co-authored-by: Ricardo Iván Vieitez Parra <3857362+corrideat@users.noreply.github.com>
Co-authored-by: Greg Slepak <contact@taoeffect.com>
Co-authored-by: snowteamer <64228468+snowteamer@users.noreply.github.com>
Co-authored-by: Sebin Song <songxevin@gmail.com>

Silver-IT committed:
feat: resolved conflicts

Silver-IT committed:
fix: errors

Silver-IT committed:
fix: error in switching chatroom as soon as it is created

Silver-IT committed:
fix: error in cypress test

corrideat committed:
Bugfixes

corrideat committed:
Fix UI inconsistency issues (wrong message when joining / leaving; switching to newly created chatroom)

corrideat committed:
Encrypt OP_KEY_SHARE envelope

SebinSong committed:
restore RecordPayment.vue against e2e-protocol branch

SebinSong committed:
work partially on Greg's CR

Silver-IT committed:
chore: pushed the latest commit of e2e-protocol-ricardo-2023

corrideat committed:
Add missing sync when added to chatroom

corrideat committed:
Remove extraneous logging

corrideat committed:
Add destructor to leave chatrooms

Silver-IT committed:
chore: merged the latest commit and fix erorr in cy.login

corrideat committed:
Leave chatrooms on removeOurselves

corrideat committed:
Remove JOINING_CHATROOM

taoeffect committed:
Merge pull request #1780 from okTurtles/e2e-protocol-ricardo-20231025-login-sync

E2e protocol ricardo 20231025 login sync

taoeffect committed:
Merge pull request #1785 from okTurtles/e2e-protocol-ricardo

E2e protocol ricardo

taoeffect committed:
Merge branch 'master' into e2e-protocol

corrideat committed:
Implement /joinGroup and /leaveGroup. TODO: Move PEK rotation (?), fix missing currentGroupId on new device

Silver-IT committed:
chore: updated the latest e2e-protocol branch

SebinSong committed:
merge the latest e2e-protocol

SebinSong committed:
restore RecordPayment.vue against e2e-protocol

SebinSong committed:
work on implementing immediate kickout permission

Silver-IT committed:
fix: cypress error

SebinSong committed:
fix the giconf.json error

corrideat committed:
Improvements to login using new actions

Silver-IT committed:
feat: remove eventHandler

Silver-IT committed:
fix: removed cy.wait

Silver-IT committed:
chore: reverted changes using browserify

corrideat committed:
Bugfixes and refactoring

corrideat committed:
Comments and fix to vuex/save error

Silver-IT committed:
feat: removed cy.wait

SebinSong committed:
create message handlers for push-notification in both client and server

corrideat committed:
Use group's CEK for joining and leaving

corrideat committed:
Fix navigation glitches when joining or creating a group

corrideat committed:
Prevent flashing of no group name

corrideat committed:
Add comments

corrideat committed:
Fix group creation buttons

corrideat committed:
Fix steps header

corrideat committed:
Fix proposals oneVoteToPass

corrideat committed:
Remove VOTE_UNDECIDED

taoeffect committed:
Merge pull request #1788 from okTurtles/e2e-protocol-ricardo

E2e protocol ricardo

Silver-IT committed:
feat: merged the latest commit and fix cypress errors

Silver-IT committed:
chore: added comments and renamed helper function

Silver-IT committed:
chore: added github discussion as comment

SebinSong committed:
add various push-action handlers to the pubsub message handler on the server side

SebinSong committed:
replace GET /push/publickey with the logic with pubsub system

SebinSong committed:
add handler logic for STORE_SUBSCRIPTION

SebinSong committed:
replace all /push/* to the pubsub equivalents

corrideat committed:
Remove return from /rotateKeys in group to avoid deadlock

corrideat committed:
Add missing import

corrideat committed:
Remove second return from call to rotateKeys

corrideat committed:
Remove /wait from rotateKeys

corrideat committed:
Chatroom: ignore join / leave side-effects when rendering messages

Silver-IT committed:
chore: merged the latest commit

SebinSong committed:
move push-subscription map to server object in createServer

corrideat committed:
Merge branch 'e2e-protocol-ricardo' into 1719-fix-all-tests-on-e2e-protocol-branch

corrideat committed:
Publish queue and reset selector

SebinSong committed:
rename TO_SERVER/FROM_SERVER to PUSH_ACTION

SebinSong committed:
Merge remote-tracking branch 'origin/e2e-protocol' into sebin/task/926-create-pwa_pubsub-integration

SebinSong committed:
some optimisation in 'push' event handler in service-worker

SebinSong committed:
correct the error sentence

SebinSong committed:
add a TODO comment in push.js

corrideat committed:
Fix user count for non-joined chatrooms

SebinSong committed:
improve the error-handling logic for push actions

corrideat committed:
Bugfixes and simplification of /join

corrideat committed:
Use on instead of once

SebinSong committed:
updates for a couple of CRs from the PR review

corrideat committed:
Remove delete pending

corrideat committed:
Add queue for listenChatoomActions

SebinSong committed:
a couple of fixes in service-worker.js

corrideat committed:
postpublish hook and internal-only id

corrideat committed:
id-less chat refreshes

corrideat committed:
Remove GIMessage.id()

corrideat committed:
Remove missing from GIMessage

corrideat committed:
Bugfix: onprocessed handler

SebinSong committed:
Remove the logic that asks server to send a test notification

SebinSong committed:
fix the undeleted logic

SebinSong committed:
separate push-subscription logic from service-worker registration

corrideat committed:
Add way for Chelonia to shut down

SebinSong committed:
move the push-setup logic into requestPermission() util method

SebinSong committed:
move '/send-push' into a then of Promise

corrideat committed:
Bugfixes for integration tests

corrideat committed:
Fix duplicate messages and move removeImmediately to internals

corrideat committed:
onresolve for forceDistributionDate

corrideat committed:
Update to commands.js

SebinSong committed:
setup/send push-notification only after the permission has been granted.

SebinSong committed:
remove console.log()

SebinSong committed:
merge the latest e2e-protocol in

taoeffect committed:
Merge pull request #1786 from okTurtles/sebin/task/#1781-immediate-memeber-kickout

#1781 - Add "Use admin permissions to remove immediately" checkbox

SebinSong committed:
logic for exchanging a message btw the SW and the Client

corrideat committed:
Fix Flow errors and make encryptedData test pass

corrideat committed:
Undo change to ChatMain

corrideat committed:
encryptedData: remove unnecessary multiple calls to rootState

corrideat committed:
encryptedData: fix selector name

corrideat committed:
Updates to make backend.test.js work; added null encryption and signature algorithms

corrideat committed:
Add comments on eNULL and sNULL as well as environment variable

SebinSong committed:
Merge remote-tracking branch 'origin/e2e-protocol' into sebin/task/926-create-pwa

SebinSong committed:
work on Greg's CR again

corrideat committed:
Write to contract state only after processing finished

corrideat committed:
Update comment

corrideat committed:
Feedback

corrideat committed:
Fix wrong selector

corrideat committed:
Various bugfixes, atomic for sendMessage

corrideat committed:
comments and deserializeHEAD

corrideat committed:
Remove unnecessary import

corrideat committed:
Fix Flow errors

corrideat committed:
Fix deserializeHEAD

corrideat committed:
Fix preSendCheck

corrideat committed:
Fix state

corrideat committed:
Duplicate proposal key prefix

corrideat committed:
fix check

corrideat committed:
Fixed race-conditions when joining a group

corrideat committed:
Improved check with both contracts

corrideat committed:
Fix removing ourselves when re-added

corrideat committed:
Add duplicate check for previous volatile state; do so only if the
contract was re-synced

corrideat committed:
Readability and fixed check

corrideat committed:
Remove unneeded logging

corrideat committed:
Fixes to chatroom

corrideat committed:
Remove logging

corrideat committed:
Merge branch 'e2e-protocol' into 1719-fix-all-tests-on-e2e-protocol-branch

taoeffect committed:
Merge pull request #1787 from okTurtles/1719-fix-all-tests-on-e2e-protocol-branch

Make Cypress test passing for E2E protocol

SebinSong committed:
work on Greg/Snowteamer Feedbacks on the PR

taoeffect committed:
Merge pull request #1770 from okTurtles/sebin/task/926-create-pwa

#926 Create a PWA for the app

SebinSong committed:
fix the white-space issue

taoeffect committed:
Merge pull request #1795 from okTurtles/sebin/task/1790-fix-add-to-channel-btn

#1790 - Remove the extra space in "Add to channel" button

corrideat committed:
Move test data attributes into Cypress commands

SebinSong committed:
move the try ... catch block into the Promise callback

taoeffect committed:
Merge pull request #1796 from okTurtles/sebin/task/fix-cypress-error-in-electron

Fix for a cypress failure related to push-subscription in Electron

corrideat committed:
Improvements to tests.

* Exclude some spec files from running in CI
* Use longer usernames to avoid collisions
* Fix bug in preSendCheck for chatroom

corrideat committed:
Re-enable tests

corrideat committed:
Merge branch 'master' into e2e-protocol-ricardo

corrideat committed:
Disable native worker temporarily to allow tests to pass

corrideat committed:
Add two tests to CI

corrideat committed:
Update test suites used

corrideat committed:
Add checks and suppress chatroom leave error

corrideat committed:
Move preSendChecks to actions. Fix: group-chat fails

corrideat committed:
Bugfixes. TODO: remove log, etc., DRY sorting in identity, fix chat-message and chat-scrolling tests

corrideat committed:
Bugfixes for pending messages; remove debug logging

snowteamer committed:
Make the pubsub system more generic (#1792)

* Remove dontBroadcast parameter

* Rename subscribersByContractID to subscribersByChannelID

* Rename contractID to channelID in pubsub

* Move defaultOptions to the top

* Emit an event upon successful sub

* Use the new sub event to sync contracts

* Fix logging of 'Already subscribed to' messages

* Rename SUCCESS to OK

* Add createOkResponse and remove createResponse

* Add basic pub implementation

* No longer broadcast UNSUB upon close event

* Replace 'contract' by 'channel' where appropriate

* Revert usage of the new ??= operator

* Improve subscription-succeeded handler

* Use force to sync contracts upon reconnection

* Disallow subscribing to unknown channels

* Remove shared/logger.js

SebinSong committed:
#1793 Dark theme status bar color in the PWA  (#1800)

* replace theme_color field in manifest.json file to the equivalent meta tag

* update status-bar color according to the theme-setting change

taoeffect committed:
Merge branch 'master' into e2e-protocol

corrideat committed:
Move General chatroom creation to side-effect, fix bug with sometimes not joining it

corrideat committed:
Add back accidentally removed tests

corrideat committed:
Merge branch 'e2e-protocol' into e2e-protocol-ricardo

corrideat committed:
Fix flow types

SebinSong committed:
#1791 - Immediately install new service worker on refresh (#1799)

* add a sbp selector for manually updating the service-worker

* call Registration.update() immediately / set up 1hr periodic timer for it

* remove useless console.log

* check WebSocket.CLOSED / WebSocket.CLOSING before sending message to the push-server

* set up a event listener for PUBSUB_RECONNECTION_SUCCEEDED sbp event

* fix the typo

* minor updates

SebinSong committed:
#1665 Typing indicator in the chatroom (#1805)

* implment [PUB] message handler in server-side pubsub

* create various pubsub related actions in chatroom.js action handlers

* make sure 'chatroom-user-typing' pubsub event works correctly

* add handler for pubsub event PUBSUB_EVENT_TYPE.CHATROOM_USER_TYPING in SendArea.vue

* delete a duplicate component method

* throttle the pubsub event-emitter function

* move the typing indicator position

* implement CHATROOM_USER_STOP_TYPING event for when the textarea field is emptied

* do not unsub/sub from the chatroom contract pubsub channel / use the existing channel to emit a message

* fix the UI bug in the chatroom / remove all console.log

* revert the block in backend/pubsub.js where it adds a channel if not exist

* detach GI specific message handlers from chelonia / pubsub

* revert the default PUB message handler to the shared.pubsub.js

corrideat committed:
Address PR comments

corrideat committed:
Bugfix: Fix group membership sync issue, re-joining us automatically

taoeffect committed:
use createCID instead of blake32hash in some places

taoeffect committed:
fix avatar upload test

corrideat committed:
Cleanup & move logic to internalSideEffectsStack (part 1)

taoeffect committed:
fixed preloading

taoeffect committed:
Merge branch 'master' into e2e-protocol

corrideat committed:
Various bugfixes and cleaning up.

- Added catch blocks to all or most free-standing promises
- Removed validation of outgoing messages; outgoing messages now
  go through processMessage instead
- Made processMessage side-effect free as much as possible
- Removed extraneous logging
- Improved the tests to verify that keys are actually rotated
- Fixed a bug in the way the invite links were displayed.
- Other smaller changes

SebinSong committed:
#1811 - Wrong file attachment detection in the chatroom (#1813)

* fix the bug

* add comment

* fix the array related bug

* remove console.log

corrideat committed:
Harmonise ChatMain scroll position with master and make topOffset a computed property

corrideat committed:
Clean up and add comments

corrideat committed:
Fixes to chatroom scrolling

corrideat committed:
Add timeout to scrolling test

corrideat committed:
Moved onChatScroll to call flush and minor improvements

corrideat committed:
Clean up and comment code

corrideat committed:
Add onChatScroll method wrapper

corrideat committed:
Remove check for null or undefined

corrideat committed:
Fixed (removed) spurious error for outgoing OP_KEY_REQUEST; fixed the way to assign globalThis.crypto

corrideat committed:
Remove unnecessary preSendCheck uses and consolidate them in contracts

corrideat committed:
Fixes for pubsub when switching sessions (reset)

corrideat committed:
Improvements for legibility

corrideat committed:
Remove pending from root state

corrideat committed:
Remove await from contracts identity.js and other small improvements

corrideat committed:
Remove await for actions in identity contract

SebinSong committed:
#1558 - markdown in chat (#1815)

* add markdown-it pkg to the project and make sure it works

* try slack-style markdown plugin

* optimise the convertToMarkdown() function

* switch from markdown-it to marked (zero-dependency pkg)

* define a custom renderer function that converts *string* to <strong>string</strong>

* style the <code> and <pre><code></code></pre> in the generated markdown outcome

* add more style declarations for markdown elements

* enable <del> tag / specify target='_blank' to <a> tag output

* add various call-to-action btns for transforming to markdown in SendArea.vue

* add a couple of more cta btns

* implement transformTextSelectionToMarkdown() method

* create injectOrStripSpecialChar() func for logic that injects special characters

* implment inject link element to the target string

* move the link cta to the last in the line

* move the primary call-to-actions next to the send btn / add style ctas to the edit mode too

* fix issue with the nested markdown styles / switch gfm on

* work on more feedback items

* delete the test file

* work on more change requests

corrideat committed:
Rewrite wait

corrideat committed:
Merge branch 'e2e-protocol-ricardo' into e2e-protocol

corrideat committed:
Disable CI tests

taoeffect committed:
Merge branch 'e2e-protocol' into hash-fix

corrideat committed:
Add esbuild optional dependencies

taoeffect committed:
Merge pull request #1810 from okTurtles/hash-fix

Hash fix

corrideat committed:
Move /login action into root component's mounted

corrideat committed:
Clean up LoginForm.vue

corrideat committed:
Improved error handling and reporting

corrideat committed:
Formatting and alerts for login errors

taoeffect committed:
Merge pull request #1820 from okTurtles/e2e-protocol-ricardo

e2e protocol ricardo

taoeffect commented on Gruntfile.js:
```diff
@@ -1,5 +1,8 @@
 
+// TODO: REMOVEME. This prevents tests from running
+if (process.env['CI']) process.exit(1)
+
```
This can now be removed

corrideat committed:
Revert "Disable CI tests"

This reverts commit 43483661fb480b4b0c706ea7d1f9c874d82eb4e7.

taoeffect commented on frontend/views/containers/chatroom/ChatMain.vue:
```diff
@@ -160,6 +227,8 @@ export default ({
   },
   data () {
     return {
+      GIMessage,
+      JSON,
```
Why are these here?

EDIT: will be cleared out in a separate PR
